rocketpy/mathutils/function.py

```diff
@@ -2857,30 +2834,55 @@ def _check_user_input(
             source_dim = source.shape[1]
 
             # check interpolation and extrapolation
+
+            ## single dimension
+            if source_dim == 2:
+                # possible interpolation values: llinear, polynomial, akima and spline
+                if interpolation is None:
+                    interpolation = "spline"
+                elif interpolation.lower() not in [
+                    "spline",
+                    "linear",
+                    "polynomial",
+                    "akima",
+                ]:
+                    warnings.warn(
+                        "Interpolation method for single dimensional functions was "
+                        + f"set to 'spline', the {interpolation} method is not supported."
+                    )
+                    interpolation = "spline"
+
+                # possible extrapolation values: constant, natural, zero
+                if extrapolation is None:
+                    extrapolation = "constant"
+                elif extrapolation.lower() not in ["constant", "natural", "zero"]:
+                    warnings.warn(
+                        "Extrapolation method for single dimensional functions was "
+                        + f"set to 'constant', the {extrapolation} method is not supported."
+                    )
+                    extrapolation = "constant"
+
+            ## multiple dimensions
```

Comment:
Shouldn't this `if source_dim > 2` be an `elif`?

Reply:
Yes, prolly yes.

I created a new condition before this if statement, didn't pay attention to this detail tks.

Reply:
Solved: https://github.com/RocketPy-Team/RocketPy/pull/470/commits/d7e4dba4970debd89088ece6a6990a934a74b35f